### ISSUE TITLE
Firefox 58.0.2 full screen problem on slideshow

### ISSUE THREAD

**Issue by yamoussouk** [SeleniumHQ/selenium#5737]
## Meta -
OS:  Windows 10
Selenium Version:  3.9.1
Geckodriver: 0.20.0
Browser:  Firefox 58.0.2 (32bit)

## Expected Behavior -

There are to options:
1. I use Linux and Intellij to run testcases on remote machine as 'local'. The hub and the node is on the remote machine. I also use Windows 7 on Oracle VirtualBox where I establish connection via Remote Desktop Connection to check testrunning on the remote machine.
2. I use Jenkins to run tests from master (linux) on slave (remote machine).
When I run selenium test on remote machine (Windows 10) from (Linux ubuntu 16.04) on firefox 58.0.2., or from Jenkins, slideshow should run on full screen which from I can escape pressing F11.

## Actual Behavior -

I use Remote Desktop Connection to connect to remote machine and watch testrunning on it. 
Slideshow runs on 366x144 resolution if and only if I don't watch the testrunning from the begining. 

![slideshow](https://user-images.githubusercontent.com/36341359/38428592-71b05fd0-39bc-11e8-8c66-c1ff0c1b221d.png)

The test fails when selenium wants to click on element which is obscured by an another element because of the small resolution.

If I click twice on firefox icon on taskbar, resolution is OK again.

**Comment by barancev** [comment]
Selenium does not manage windows, the operating system (or, window manager) does. So it must be an issue in Remote Desktop Connection client that does not change browser window resolution after connection. Can you reproduce the issue if you connect to remote host using the same resolution it already has (366x144)? Can you reproduce the issue if you simply set small browser window size on the local machine?

**Comment by yamoussouk** [comment]
Hello.

Thank you for your answer.

I use Remote Desktop Connection (RDC) only for checking testrunning. It is not relevant for the issue. If I don't watch testrunning via RDC the resolution problem occurs anyway. The window resolution is printed on the console, it is 366x144 instead of FHD. 

It is something with firefox 58.0.2 or higher or with selenium. I tried on Chrome 64 and 65, and there is no problem with the window resolution. I have just upgraded from selenium 2.53 and firefox 47.0.2 to selenium 3 and firefox 58.0.2. With selenium 2 there was no problem with slideshow so far.

I tried to maximize the window size if it equals to 366, but in this case slideshow exits from full screen mode.

I can try out your suggestions at the earliest moment on Monday.

**Comment by cgoldberg** [comment]
> I tried to maximize the window size if it equals to 366, 
> but in this case slideshow exits from full screen mode.

Note that maximizing is different than fullscreen. 

Also, how are you entering fullscreen mode? (you didn't mention which bindings you are using or give any example code)

**Comment by yamoussouk** [comment]
Hello
"Also, how are you entering fullscreen mode? (you didn't mention which bindings you are using or give any example code)"
We are using window.fullscreen
https://developer.mozilla.org/en-US/docs/Web/API/Window/fullScreen

Here is a link with a shared gallery: https://www.irista.com/gallery/ekljmp7rais4
At the right top corner of the screen click on the 3 dot then play slideshow. When the slideshow starts it changes to full screen mode. As the photo shows above in our testrunning on firefox 58.0.2. the slideshow's window resolution is 366x144 (obviously we are using testenvironments).

**Comment by cgoldberg** [comment]
@yamoussouk you still haven't mentioned which bindings you are using or given any example code.

> We are using window.fullscreen

`Window.fullScreen` is just a property that indicates whether the window is displayed in full screen mode or not.

**Comment by yamoussouk** [comment]
UPDATE:
Since my last comment I've realized that every test case is involved in this problem. Fullscreen was just top of the iceberg. 

When I start a test case in Firefox 58.0.2 or higher the browser window size is fullHD (1920x1080), but it uses lower media query (500px screen width), therefore elements -which are visible on fullHD- are not visible on low query.

We use driver.manage().window().maximize() to maximize the window size. I tried to use setPosition(new Point(0,0)) and setSize(new Dimension(1920, 1080)) or i tried FirefoxOptions() and to addArguments("--start-maximized") or addArguments("-width 1920", "height 1080") but it did not work. It sized the window to fullHD, but the media query was still low.

When I use window().fullScreen() the browser window size is 366x144... It shows the fullscreen page but in that small window.

In conclusion, Firefox or Geckodriver "thinks" that its size is lower than 500px, still it resizes itself to fullHD.

**Comment by yamoussouk** [comment]
Thank you for your answer. We test scenarios on Firefox and Chrome. As I mentioned before Chrome is running tests on appropriate way, the browser size is 1920x1080, and the query refers to it, the element is visible as expected.

**Comment by yamoussouk** [comment]
I am running these test cases on a PC without monitor. Does it cause the problem? There is no set resolution, therefore window manager can't provide the relevan resolution/size?!

**Comment by yamoussouk** [comment]
UPDATE:
Headless mode solved the problem temporarily. As far as I know virtual browser is equivalent to "real" browser.

**Comment by barancev** [comment]
No idea why this issue is still open here. If the issue is still actual it should be reported to https://github.com/mozilla/geckodriver/issues
